.github/workflows/ci.yml

```diff
@@ -0,0 +1,132 @@
+
+# This file was created in part or in whole by one of OpenAI's generative AI models
+
+name: Continuous Integration
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  ci_format_job:
+    if: >
+      ${{ !contains(github.event.pull_request.title, 'Draft:') &&
+          !contains(github.event.pull_request.title, 'WIP:') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Run format check
+        run: |
+          source env/bash
+          VERBOSE=1 ./style/format.sh
+          git diff --exit-code --ignore-submodules
+
+  ci_cpu_job:
+    if: >
+      ${{ !contains(github.event.pull_request.title, 'Draft:') &&
+          !contains(github.event.pull_request.title, 'WIP:') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up environment
+        run: |
+          source env/bash
+      - name: Run CPU tests
+        run: |
+          cd tst
+          python3 run_tests.py regression.suite \
+            --save_build \
+            --make_nproc=32 \
+            --cmake=-DCMAKE_C_COMPILER=gcc \
+            --cmake=-DCMAKE_CXX_COMPILER=g++ \
+            --log_file=ci_cpu_log.txt
+      - name: Upload CPU test log
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ci_cpu_log.txt
+          path: tst/ci_cpu_log.txt
+          retention-days: 3
+      - name: Upload figures
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: figs
+          path: tst/figs
+          retention-days: 3
+
+  ci_gpu_job:
+    if: >
+      ${{ !contains(github.event.pull_request.title, 'Draft:') &&
+          !contains(github.event.pull_request.title, 'WIP:') }}
+    runs-on: ubuntu-latest  # Update to a runner with GPU support if available
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up environment
+        run: |
+          source env/bash
+      - name: Run GPU tests
+        run: |
+          cd tst
+          python3 run_tests.py gpu.suite \
+            --save_build \
+            --make_nproc=32 \
+            --cmake=-DARTEMIS_ENABLE_CUDA=On \
+            --cmake=-DKokkos_ARCH_VOLTA70=On \
+            --cmake=-DCMAKE_CXX_COMPILER=external/parthenon/external/Kokkos/bin/nvcc_wrapper \
+            --log_file=ci_gpu_log.txt
+      - name: Upload GPU test log
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ci_gpu_log.txt
+          path: tst/ci_gpu_log.txt
+          retention-days: 3
+      - name: Upload figures
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: figs
+          path: tst/figs
+          retention-days: 3
+
+  ci_doc_job:
```

Comment:
I don't think this one is needed given `docs.yml`?

Reply:
Do we want to split up building and deploying docs? E.g. do we want to check that docs build for every PR, but only deploy the docs whenever something gets merged into `develop`?

Reply:
That's how it's setup currently. It will only deploy on a push to develop. An MR will just build.

Reply:
Ah I see it now, OK I removed all docs stuff from the CI